### ISSUE TITLE
Please register custom URL scheme 'app...'in the app's info.plist file (IOS)

### ISSUE THREAD

**Issue by lxrinc** [FlutterFlow/flutterflow-issues#1373]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I am trying to login on IOS using test flight and receiving an error when I press the sign-in button. A snack bar shows up with the error "Please register custom URL scheme 'app...'in the app's info.plist file (IOS)".

### Expected Behavior

I expected the screen to redirect to the phone verification screen where a verification code is sent. It works very well on Android devices but IOS, it is giving an error.

### Steps to Reproduce

https://github.com/FlutterFlow/flutterflow-issues/assets/117313181/8c0c1308-f38a-41f6-85ae-0e4b80927751

1. Access the App
2. Redirects to choose a language
3. Select on the "Get Started" button
4. Choose a country code and any phone number
5. Press Sign In
6. The error shows up

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITEehcmA0492pbwA0LrcK8FslQMwJ0QlUIInj+0bWRMhNZfpEZYAf+DSTBFLYtTne25+C1WYhnkC0PTzkezxGvYEHxedfJxZ+b5XQQ7heDqlbK2VDqWwOnBCQJ1XJ2Kz0Z7Qvx0mAM1udVYG6mePevCdG3qCf9qOYwx5e6fDbOY=

### Context

I am trying to test my app on iphone devices but after the login screen I cannot proceed to test the app.

### Visual documentation

https://github.com/FlutterFlow/flutterflow-issues/assets/117313181/99b86bbc-7963-4224-8ad4-e6a1e28260e7

### Environment

```markdown
- FlutterFlow version: v3.1 (released 28th August, 2023)
- Platform: Web
- Browser name and version: Brave - Version 1.57.53 Chromium: 116.0.5845.114 (Official Build) (64-bit)
- Operating system and version affected: Windows 10
```

**Comment by hariprasadms** [comment]
Hi @lxrinc - Thanks for submitting the issue. I looked at your project downloaded code and I can reproduce above error message. However, in general plist property details related to url schema it looks all right. See below. Not really sure how your project got this error. 

<img width="1509" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/17435678/f400fa5e-71db-40c7-86bc-0e753de06783">

I would suggest, re-built the iOS version one or two more times and see consistency of the issue. If issue still persists, I can mark this as project specific issue. Let me know how it goes. Thanks

**Comment by lxrinc** [comment]
@hariprasadms I have deployed around 7 times but the issue is persisting. It could be a project-specific issue, how can it be resolved from flutterflow?

**Comment by lxrinc** [comment]
![1 25 Build 28](https://github.com/FlutterFlow/flutterflow-issues/assets/117313181/de3333a7-1fdc-4abf-a571-9cfb334e6310)
![5 14](https://github.com/FlutterFlow/flutterflow-issues/assets/117313181/b539e35c-962b-486d-ae02-97eca0eb9918)
![7 25](https://github.com/FlutterFlow/flutterflow-issues/assets/117313181/78228f45-5a3d-4013-b2bd-3553880da0a4)
![1 22 Build 27](https://github.com/FlutterFlow/flutterflow-issues/assets/117313181/62298973-7ee9-438d-8eec-938fd05a1cc8)

Above are 4 different builds and showing the same error, this excludes the build with the video.

**Comment by lxrinc** [comment]
@hariprasadms I cross-checked with Email Auth and it works but with Phone Auth it is still persisting the error.

**Comment by hariprasadms** [comment]
Hi, Sorry, I looked your project again but I see _CFBundleURLSchemes_ value is not populated. I think this could be the issue. Can you please check and add this value manually from downloaded _**GoogleService-info.plist**_ file

Sample - 
<img width="1095" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/17435678/41e12a57-7026-4d33-a6ef-1aac0a6ac715">

Below is your project info.plist file.
<img width="1512" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/17435678/75d42044-741c-486e-a21d-1042a4b6f10c">

**Comment by lxrinc** [comment]
Hi, @hariprasadms I regenerated the Firebase config, and yet the error is persisting. I will wait for your feedback.

**Comment by jasminder** [comment]
@hariprasadms This issue is persisting. Kindly look into this. 
cc @agreaves

**Comment by adialloo** [comment]
same problem

**Comment by julien-digilink** [comment]
same problem

**Comment by sbis04** [comment]
Everyone facing this issue, if you download the `GoogleServiceInfo.plist` file from the Firebase console (it should be inside gear icon(left panel)> Project settings > General > scroll down and select the iOS app), does it contain the `REVERSED_CLIENT_ID` field?

<img width="1668" alt="Screenshot 2023-09-19 at 1 59 28 PM" src="https://github.com/FlutterFlow/flutterflow-issues/assets/43280874/39bceb50-b740-4948-ae23-96f9b3731b0f">

**Comment by lxrinc** [comment]
@sbis04 It doesn't contain the REVERSED_CLIENT_ID in the GoogleServiceInfo.plist

**Comment by sbis04** [comment]
@lxrinc Okay. And, just making sure, in Firebase Authentication which auth methods do you have enabled (is it only Phone Auth)?

**Comment by sbis04** [comment]
Hmm, this is interesting. Not sure why `REVERSED_CLIENT_ID` is not present in the file, I'll investigate into it further.

@lxrinc  One more question, did you try `Regenerate Config File` from `Settings > Firebase` inside FlutterFlow? Did that still not fix the issue?

**Comment by lxrinc** [comment]
@sbis04 Yes I did that a few times, assuming that was the problem but it still persisted.

**Comment by sbis04** [comment]
@lxrinc Can you try these steps once?

1. Enable "Google Sign In" in your Firebase project.
2. Go to the Firebase console > Project settings > Select the iOS app > Remove this app.
3. Regenerate the config file again inside FlutterFlow
4. Try the app or check if the `GoogleService-Info.plist` has the `REVERSED_CLIENT_ID` field now.

Let me know if this helps in any way.

**Comment by sbis04** [comment]
Nice, thanks for letting me know! This seems like some kind of bug on the Firebase side, because phone auth requires the `REVERSED_CLIENT_ID` still it doesn't get generated unless Google Sign In is enabled in some cases.

**Comment by DamilolaDami** [comment]
can this be fixed because this requires us to do some extra config that our apps doesn't really needs. Thank you

**Comment by Gintasz** [comment]
> can this be fixed because this requires us to do some extra config that our apps doesn't really needs. Thank you

try to enable email & google sign in on project firebase, even if you don't use it, that fixed this error for me.

**Comment by DamilolaDami** [comment]
> that

Yeah this is a workaround, not a solution

**Comment by DamilolaDami** [comment]
> > can this be fixed because this requires us to do some extra config that our apps doesn't really needs. Thank you
> 
> 
> 
> try to enable email & google sign in on project firebase, even if you don't use it, that fixed this error for me.

Yea this a workaround, not a fix.

**Comment by eujoseant** [comment]
> > can this be fixed because this requires us to do some extra config that our apps doesn't really needs. Thank you
> 
> try to enable email & google sign in on project firebase, even if you don't use it, that fixed this error for me.

Do I have to keep these options turned on afterwards? Even if I don't use them? For better operation and not generating the error or I can turn it off.

**Comment by navaidali** [comment]
Getting same error, I tried to regenerate the config file but the issue persists. any solution?

**Comment by ublanke** [comment]
The above workaround did not work for me (including the REVERSED_CLIENT_ID variable).
Also, this is not a fixed issue, if users active phone only. It should not be closed, nor marked as 'not a bug'.

**Comment by eyuphan-oguz** [comment]
Copy the value inside the single quotes from the Snackbar, for example: 'app-1-123456789-ios-bcbaac540bd7dae7a'.

Open your Flutter project.

Locate the ios folder in your project directory.

Right-click on the ios folder and choose "Open in Xcode".

In Xcode, find the name of your project on the left, right-click on it, and choose "Open in Finder".

In the Finder window, locate the "Runner" folder.

Inside the "Runner" folder, double-click to open the Info.plist file.

In the opened Info.plist file, switch to the "Information Property List" view at the top.

To add URL Types, right-click on an empty space, and select "Add Row".

In the added row, go to the left column and name it "URL types".

In the right column of the added row, click on the dropdown arrow and create "Item 0" underneath.

Under "Item 0", set URL Identifier to "app-1-123456789-ios-bcbaac540bd7dae7a".

Under "Item 0", set URL Schemes to "app-1-123456789-ios-bcbaac540bd7dae7a".

Save your changes and close Xcode.

**Comment by sai-chandra22** [comment]
This REVERSE_CLIENT_ID is creating the reCAPTCHA issue, how to remove reCAPTCHA @jasminder @Gintasz 
@julien-digilink @hariprasadms

**Comment by alih552** [comment]
I've tried all these steps:
1- Enable "Google Sign In" in your Firebase project.
2- Go to the Firebase console > Project settings > Select the iOS app > Remove this app.
3- Regenerate the config file again inside FlutterFlow

Problem is I am getting an unknown error when trying to regenerate config files after the ios app is removed from firebase.

@sbis04 @kevintonb

**Comment by alih552** [comment]
Okay, I solved the issue by permanently deleting the app. After deleting it, it still exists and requires 29 days to be truly deleted.

**Comment by kanha-d** [comment]
Hola, I resolved the issue by adding GOOGLE_APP_ID - 1:445104433345:ios:b65a404688719bf4f50c40

in CFBundleIdentifier & CFBundleURLSchemes with app-1-GOOGLE_APP_ID

please check below schema and run your app again 

<dict>
			<key>CFBundleIdentifier</key>
			<string>1:445104433345:ios:b65a404688719bf4f50c40</string>
			<key>CFBundleTypeRole</key>
			<string>Editor</string>
			<key>CFBundleURLSchemes</key>
			<array>
				<string>app-1-445104433345-ios-b65a404688719bf4f50c40</string>
			</array>
		</dict>
